disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,209 @@
+
+    station_name = metadata_dict.get("station_name")
+
+    if station_name and station_name != expected_station_name:
+        return None, None
+
+    # Get data url
+    station_remote_url = metadata_dict.get("data_url")
+
+    # Get the local path
+    data_dir_path = os.path.dirname(yaml_file_path).replace("metadata", "data")
+    station_dir_path = os.path.join(data_dir_path, station_name)
+
+    return station_dir_path, station_remote_url
+
+
+def _get_local_and_remote_data_directories(
+    disdrodb_dir: str,
+    data_sources: Optional[Union[str, List[str]]] = None,
+    campaign_names: Optional[Union[str, List[str]]] = None,
+    station_names: Optional[Union[str, List[str]]] = None,
+) -> list:
+    """Parse the folder according to the parameters (data_source,
+    campaign_name and station_name) to get all url from the config files.
+
+    Parameters
+    ----------
+    disdrodb_dir : str , optional
+        DisdroDB data folder path.
+        Must end with DISDRODB.
+    data_sources : str or list of str, optional
+        Data source folder name (eg : EPFL).
+        If not provided (None), all data sources will be downloaded.
+        The default is data_source=None.
+    campaign_names : str or list of str, optional
+        Campaign name (eg :  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be downloaded.
+        The default is campaign_name=None.
+    station_names : str or list of str, optional
+        Station name.
+        If not provided (None), all stations will be downloaded.
+        The default is station_name=None.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+
+    # Get all config files from the metadata folders
+    list_of_base_path = []
+    if data_sources:
+        if isinstance(data_sources, str):
+            data_sources = [data_sources]
+    else:
+        data_sources = ["**"]
+    if campaign_names:
+        if isinstance(campaign_names, str):
+            campaign_names = [campaign_names]
+    else:
+        campaign_names = ["**"]
+
+    for data_source in data_sources:
+        for campaign_name in campaign_names:
+            base_path = os.path.join(disdrodb_dir, "Raw", data_source, campaign_name)
+            list_of_base_path.append(base_path)
+
+    metadata_folder_name = "metadata"
+
+    list_metadata_folders = []
+
+    for base_path in list_of_base_path:
+        if station_names:
+            if isinstance(station_names, str):
+                station_names = [station_names]
+            for station_name in station_names:
+                metadata_path = os.path.join(base_path, "**", metadata_folder_name, f"{station_name}.yml")
+                list_metadata_folders += glob.glob(metadata_path, recursive=True)
+        else:
+            metadata_path = os.path.join(base_path, "**", metadata_folder_name, "*.yml")
+            list_metadata_folders += glob.glob(metadata_path, recursive=True)

```

Comment:
This should go into the `get_metadata_list` functions in `api.metadata.py.
You could add an argument `with_stations_data=True` to use current function, and when `False` use the one you implemented here ;) @regislon

Reply:
ok